profiling/src/lib.rs

```diff
@@ -168,9 +165,10 @@ pub extern "C" fn get_module() -> &'static mut zend::ModuleEntry {
         post_deactivate_func: Some(prshutdown),
         deps: DEPS.as_ptr(),
         ..Default::default()
-    };
+    });
 
-    Box::leak(Box::new(module))
+    // SAFETY: well, it's as least as safe as what every single C extension does.
+    unsafe { &mut *MODULE }
```

Comment:
Just for my understanding:
`Box::new` allocated on the heap and `Box::leak` made sure, that Rust does not care anymore.
With this change to `Lazy` this will now return a mut pointer to this local static and Rust will take care and free this space in the shutdown of the program to not trigger ASAN.
Correct?

Reply:
Close. Since it's a local static, it's never freed. It just lives in static memory forever, but this isn't considered a leak, just like having a static `int` isn't a leak.

The main difference aside from the leaking is that before we'd return a new module for every call to `get_module` but now we return the same module. It's expected to only be called once anyway, and the engine will yell at you if you load a module more than once.

But this is doing what every C module does: returning a static module.